packages/pods-core/src/builtin/OrbitPod.ts

```diff
@@ -0,0 +1,350 @@
+          orbitId,
+          social: {
+            linkedin,
+            github,
+            twitter,
+            discord,
+            hn,
+            website,
+          },
+        },
+      };
+      let noteName = name || github || discord;
+      noteName = DNodeUtils.cleanFname(noteName);
+      let fname;
+
+      const note = NoteUtils.getNoteByFnameV5({
+        fname: `people.${noteName}`,
+        notes: engine.notes,
+        vault,
+        wsRoot,
+      });
+
+      if (!_.isUndefined(note)) {
+        if (!_.isUndefined(note.custom.orbitId)) {
+          const conflictData = this.getConflictedData({ note, member });
+          if (conflictData.length > 0) {
+            fname = `people.orbit.duplicate.${Time.now().toFormat(
+              "y.MM.dd"
+            )}.${noteName}`;
+            conflicts.push({
+              conflictNote: note,
+              conflictEntry: NoteUtils.create({ ...member, fname, vault }),
+              conflictData,
+            });
+          } else {
+            await this.updateNoteData({ note, member, engine });
+          }
+        }
+      } else {
+        fname = `people.${noteName}`;
+        create.push(NoteUtils.create({ ...member, fname, vault }));
+      }
+    });
+
+    return { create, conflicts };
+  }
+
+  /**
+   * returns all the conflicted entries in custom.social FM field of note
+   */
+  getConflictedData = (opts: { note: NoteProps; member: any }) => {
+    const { note, member } = opts;
+    const customKeys = Object.keys(member.custom.social);
+    return customKeys.filter((key: string) => {
+      return (
+        note.custom.social[key] !== null &&
+        member.custom.social[key] !== note.custom.social[key]
+      );
+    });
+  };
+
+  /**
+   * updates the social fields of a note's FM
+   */
+  updateNoteData = async (opts: {
+    note: NoteProps;
+    member: any;
+    engine: DEngineClient;
+  }) => {
+    const { note, member, engine } = opts;
+    const customKeys = Object.keys(member.custom?.social);
+    let shouldUpdate = false;
+    customKeys.forEach((key: string) => {
+      if (
+        note.custom?.social[key] === null &&
+        member.custom.social[key] !== null
+      ) {
+        note.custom.social[key] = member.custom.social[key];
+        shouldUpdate = true;
+      }
+    });
+    if (shouldUpdate) {
+      await engine.writeNote(note, { updateExisting: true });
+    }
+  };
+
+  async onConflict(opts: {
+    conflicts: Conflict[];
+    index: number;
+    handleConflict: (
+      conflict: Conflict,
+      conflictResolveOpts: PodConflictResolveOpts
+    ) => Promise<string | undefined>;
+    engine: DEngineClient;
+    conflictResolvedNotes: NoteProps[];
+    conflictResolveOpts: PodConflictResolveOpts;
+  }): Promise<any> {
+    const {
+      conflicts,
+      handleConflict,
+      engine,
+      conflictResolvedNotes,
+      conflictResolveOpts,
+    } = opts;
+    let { index } = opts;
+    const conflict = conflicts[index];
```

Comment:
why are you using an `index` instead of just iterating through conflicts? is it to handle one conflict at a time? if so, you can use the following pattern which is more readable

```ts
for (const conflict of conflicts) {
  await ...
}
```

Reply:
we are using an index here to handle one conflict at a time (sequentially).